### PR TITLE
[2/n] Fix rich model summary for tuples

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Related #9647 

RichModuleSummary crashed whenever the input/output sizes were tuples. This ensures we convert them to strings before passing them to the console for rich logging.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/rich_model_summary.py [hunk 1]
```diff
@@ -88,6 +88,8 @@ def summarize(
                 table.add_column(column_name, justify="right", style="white")
 
         rows = list(zip(*(arr[1] for arr in summary_data)))
+        # convert all elements to str for Rich
+        rows = list(list(str(x) for x in row) for row in rows)
```
Comment (kaushikb11) [inline pytorch_lightning/callbacks/rich_model_summary.py:92]: Should we convert it to str before passing it to the `summarize` method?

Comment (SeanNaren) [reply pytorch_lightning/callbacks/rich_model_summary.py:92]: That's a good idea, I just looked further down the trace and it seems we don't really enforce that the input/output sizes are strings. I'll modify this PR to reflect this instead!

**Review by kaushikb11 — APPROVED** [review]
LGTM!

File: tests/callbacks/test_rich_model_summary.py [hunk 2]
```diff
@@ -11,25 +11,52 @@
...
+        @property
+        def example_input_array(self) -> Any:
+            return torch.randn(4, 32)
+
+    model = TestModel()
+    summary = summarize(model)
+    summary_data = summary._get_summary_data()
+
+    model_summary.summarize(summary_data=summary_data, total_parameters=1, trainable_parameters=1, model_size=1)
+    assert mock_console.call_count == 2
```
Comment (carmocca) [inline tests/callbacks/test_rich_model_summary.py:59]: why 2?

**Review by carmocca — APPROVED** [review]

File: tests/callbacks/test_rich_model_summary.py (hunk 2 above)
Comment (SeanNaren) [reply tests/callbacks/test_rich_model_summary.py:59]: I'll add a comment, its called twice

File: tests/callbacks/test_rich_model_summary.py [hunk 3]
```diff
@@ -11,25 +11,54 @@
...
-
     if not _RICH_AVAILABLE:
         with pytest.raises(ImportError, match="`RichModelSummary` requires `rich` to be installed."):
             Trainer(callbacks=RichModelSummary())
+
+
+@RunIf(rich=True)
+@mock.patch("pytorch_lightning.callbacks.rich_model_summary.Console.print", autospec=True)
+@mock.patch("pytorch_lightning.callbacks.rich_model_summary.Table.add_row", autospec=True)
+def test_rich_summary_tuples(mock_table_add_row, mock_console):
```
Comment (awaelchli) [inline tests/callbacks/test_rich_model_summary.py:45]: @SeanNaren shouldn't a test be added to the model summary utility tests?

Perhaps there was a confusion here, there are different ModelSummaries, one is a callback the other is a normal class from utilities:

```
pytorch_lightning.utilities.model_summary.ModelSummary
pytorch_lightning.callbacks.model_summary.ModelSummary
```

I would argue this change should be tested universally for the utility ModelSummary

Comment (SeanNaren) [reply tests/callbacks/test_rich_model_summary.py:45]: thanks @awaelchli yes it should be, was curious if this should be tested through the callback ModelSummary or through the utility as it's a protected function. I much prefer just testing the protected function as this is much simpler

**Review by tchaton — APPROVED** [review]
LGMT !
